packages/subgraph/src/packages/token/token-voting.ts

```diff
@@ -228,3 +230,16 @@ export function handleVotingSettingsUpdated(
     packageEntity.save();
   }
 }
+
+export function handleMembershipContractAnnounced(
+  event: MembershipContractAnnounced
+): void {
+  let token = event.params.definingContract;
+  let packageEntity = TokenVotingPlugin.load(event.address.toHexString());
+
+  if (packageEntity) {
+    packageEntity.token = handleERC20Token(token);
```

Comment:
If our contracts can determine the difference, the subgraph can do it as well. 

ERC165 (which is in the ERC721 standard) can be used.

We have more means to distinguish the two because of 
GovernanceERC20, GovernanceWrappedERC20 that we control
and
GovernanceERC721 (GovernanceWrappedERC721) that we will provide.

However, the subgraph needs to change when we implement the NFT support, that's for sure.

Reply:
As a shortterm fix, we can

**a.)** leave it as it is and don't expect the admin address to be changed by manual permission granting / revoking.
**b.)** store the admin address on the plugin implementation contract and do it the `onlyOwner` way.

I would go with **a.)** for the sake of time and for the argument of "using our permission manager".
Moreover, this is a plugin. We can release a new version so let's keep it as it is

Reply:
In other words, is this future proof in terms of identifying?

Reply:
> In other words, is this future proof in terms of identifying?

Not sure if I understand your question. If we talk about the event being emitted (this is what the task is about): yes, **the contract side is future-proof**, yes.
The data emitted in `MembershipContractAnnounced`, in this case the address of a token, is sufficient / future-proof to handle NFTs later on and discriminate the two on the subgraph.

As I see it, the subgraph is not future-proof right now. We haven't thought about an NFT token type. Probably our `schema.graphql` needs to change a bit.


In `function handleMembershipContractAnnounced`, we need to distinguish between the token types and fill the respective fields.